chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredictionService.java

```diff
@@ -1,11 +1,10 @@
 package software.bytepushers.pick3.services;
 
+import com.amazonaws.services.lambda.invoke.LambdaFunction;
 import software.bytepushers.pick3.api.v1.DrawingTime;
 
 import java.time.LocalDate;
 
 public interface Pick3PredictionService {
-
-    int[][] predictPick3Plays(Integer winningNumber, LocalDate winningDrawDate, DrawingTime winningDrawTime,
-                              LocalDate futureDrawDate, DrawingTime futureDrawTime, Integer numberOfPlaysToPredict);
+    @LambdaFunction(functionName="bytepushers-chucks-pick3-lotto-system") int[][] generatePredictions(int winningNumber);

```

Comment:
This seems weird. Pick3PredictionService is simultaneously serving two roles, breaking the single-responsibility principle.

1) It is defining the signature of the AWSLambda function in order to provide the lambda function as an RPC to our web service.
1) It is defining the signature of the Pick3PredictionService which Pick3PlaysService uses to implement its call.

These two functions should be implemented as two separate interfaces / services.

Reply:
Yeah I kind of feel the same way.